### ISSUE TITLE
[BUG][Snowflake External Table] Not refreshed automatically

### ISSUE THREAD

**Issue by vamshigv** [delta-io/delta#2007]
### Describe the problem

Would like to query delta tables using Snowflake by creating external tables like mentioned [here](https://docs.snowflake.com/en/sql-reference/sql/create-external-table). Following the documentation described [here](https://docs.delta.io/latest/snowflake-integration.html) i.e. using manifest files.

#### Steps to reproduce

1. Create a Delta table with continuous updates.
2. Create Snowflake Delta External table and here is a sample SQL reference 
`create or replace stage some_stage url='s3://my-bucket/path-to-delta-lake-root' STORAGE_INTEGRATION = snowflake_access_integration;
CREATE OR REPLACE EXTERNAL TABLE delta_manifest_table(
    filename VARCHAR AS split_part(VALUE:c1, '/', -1)
)
LOCATION = @some_stage/_symlink_format_manifest/
FILE_FORMAT = (TYPE = CSV)
PATTERN = '.*[/]manifest'
AUTO_REFRESH = true;

select * from delta_manifest_table

CREATE OR REPLACE EXTERNAL TABLE some_test_table(
    id INT AS (VALUE:id::INT),
    parquet_filename VARCHAR AS split_part(metadata$filename, '/', -1)
  )
  WITH LOCATION = @some_stage/
  FILE_FORMAT = (TYPE = PARQUET)
  PATTERN = '.*[/]part-[^/]*[.]parquet'
  AUTO_REFRESH = true;

select * from some_test_table

CREATE OR REPLACE VIEW my_delta_table AS
    SELECT id
    FROM some_test_table
    WHERE parquet_filename IN (SELECT filename FROM delta_manifest_table);

  SELECT COUNT(*) FROM my_delta_table`
#### Observed results

Delta table updates aren't refreshed hence not queryable via Snowflake

#### Expected results

Delta table updates should be reflected when queried via snowflake.

**Comment by allisonport-db** [comment]
Are you updating the manifests? https://docs.delta.io/latest/snowflake-integration.html#step-3-update-manifests

**Comment by vamshigv** [comment]
Yes manifests are updated and I verified contents as well. But on the Snowflake side none of those changes are reflected

**Comment by moredatapls** [comment]
@vamshigv have you tried the newer and improved Delta integration? I've been using it for some time without any issues https://docs.snowflake.com/en/user-guide/tables-external-intro#delta-lake-support

**Comment by vgudavarthi** [comment]
@moredatapls Yes, I tried that but had problems with auto-refresh. Can you please help me understand how you were triggering the auto-refresh of tables with this specific method?

**Comment by vamshigv** [comment]
@moredatapls Yes, I tried that but had problems with auto-refresh. Can you please help me understand how you were triggering the auto-refresh of tables with this specific method?

**Comment by moredatapls** [comment]
@vamshigv I don't think that this is possible easily with external tables in Snowflake. In the article I linked it says:

> Note that the ordering of event notifications triggered by DDL operations in cloud storage is not guaranteed. Therefore, the ability to automatically refresh the metadata is not available for external tables that reference Delta Lake files. Instead, periodically execute an [ALTER EXTERNAL TABLE … REFRESH](https://docs.snowflake.com/en/sql-reference/sql/alter-external-table) statement to register any added or removed files.

We are using external Delta tables in Snowflake and how we solved it is by issuing an `ALTER EXTERNAL TABLE REFRESH` command from the Spark job that updates the Delta table using the [Snowflake JDBC driver](https://docs.snowflake.com/en/developer-guide/jdbc/jdbc). It's not ideal, but it was the easiest solution we found. If you have consumers that read from your external table you could perhaps even issue REFRESH commands there, before you consume the data?

Snowflake has a (in my opinion) [crazy complicated way to auto-refresh external tables from various cloud storages](https://docs.snowflake.com/en/user-guide/tables-external-auto), but I haven't tried it to be honest, and I am not sure if it supports Delta.

**Comment by vamshigv** [comment]
thanks @moredatapls.

**Comment by foss-contributor** [comment]
@vamshigv, thank you for your contribution! This issue has been inactive for over 18 months and will be closed in one week. To keep it open, please leave a comment on this issue or contact us in the `#contributors` channel on [Delta Lake Slack](https://go.delta.io/slack). We appreciate your involvement in Delta Lake!
